core-rust/state-manager/src/metrics.rs

```diff
@@ -330,6 +344,202 @@ impl CommittedTransactionsMetrics {
     }
 }
 
+impl ProtocolMetrics {
+    pub fn new(registry: &Registry, initial_protocol_state: &ProtocolState) -> Self {
+        let instance = Self {
+            protocol_update_readiness_ratio: GaugeVec::new(
+                opts(
+                    "protocol_update_readiness_ratio",
+                    "A ratio of supporting stake to total stake in the current validator set.",
+                ),
+                &["readiness_signal_name"],
+            )
+            .registered_at(registry),
+            pending_update_threshold_required_ratio: GaugeVec::new(
+                opts(
+                    "protocol_update_pending_threshold_required_ratio",
+                    "Required readiness ratio for the given protocol update threshold.",
+                ),
+                &["protocol_version_name", "readiness_signal_name", "threshold_index"],
```

Comment:
Maybe we should include a `threshold_epochs` label too in these :)

Reply:
hmh it's already there as a separate gauge, not sure why we would need it as a label?

Reply:
Fair - I was just thinking if you want a single graph of these lines, along with the current ratios, being able to label with the threshold epochs is probably more informative than the threshold index

Reply:
ah, gotcha :+1: okay, I'll add it